### PR TITLE
feat: 🎸 Add vertualized list to fix `CallList` performance prob

### PR DESCRIPTION
**Pull request by futurechallenger**
Add virtualized list to fix `CallList` performance problem

BREAKING CHANGE: In previous `CallItem` state are lifted to `CallList`. But this should
not cause any problems.

### CODE REVIEW

File: packages/ringcentral-widgets/components/CallsListPanel/index.js [hunk 1]
```diff
@@ -366,14 +393,55 @@ export default class CallsListPanel extends Component {
...
+      <SpinnerOverlay />
+    ) : (
+      <div
+        className={classnames(styles.list, className)}
+        ref={this._callListWrapper}
+      >
+        <div className={styles.listTitle}>
+          {onlyHistory ? null : i18n.getString('historyCalls', currentLocale)}
+        </div>
+        {useNewList ? (
```
Comment (u9520107) [inline packages/ringcentral-widgets/components/CallsListPanel/index.js]: I'd recommend using a
```javascript
const callList= useNewList ? (...) : (...);
```
block before this to make this markup section more readable, or even put it into a renderCallList function if that makes sense.

File: .gitignore [hunk 2]
```diff
@@ -60,3 +60,5 @@ alias.json
 */**/*.log
 
 */**/gh-pages
+
+.prettierrc
```
Comment (u9520107) [inline .gitignore]: I'd suggest committing this as well, perhaps we can workout a good prettier setup that can be shared across teams.
Please also add ```*.prettierrc``` entry to ```.gitattributes``` as well.

File: packages/ringcentral-widgets/components/CallsListPanel/index.js [hunk 3]
```diff
@@ -175,6 +181,22 @@ export default class CallsListPanel extends Component {
     this.forceUpdate();
   }
 
+  // componentDidUpdate(prevProps, prevState) {
```
Comment (u9520107) [inline packages/ringcentral-widgets/components/CallsListPanel/index.js]: Please delete unwanted code sections like this, we can always use git commit history to get these code back.

File: packages/ringcentral-widgets/components/CallsListPanel/index.js (hunk 1 above)
Comment (futurechallenger) [reply packages/ringcentral-widgets/components/CallsListPanel/index.js]: done

File: .gitignore (hunk 2 above)
Comment (futurechallenger) [reply .gitignore]: done

Comment (futurechallenger) [reply .gitignore]: done

**Pull request merged**